devicetypes/smartthings/aeotec-doorbell-6.src/aeotec-doorbell-6.groovy

```diff
@@ -0,0 +1,214 @@
+}
+
+def parse(String description) {
+	def result = null
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		result = zwaveEvent(cmd)
+	}
+	log.debug "Parse returned: ${result.inspect()}"
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	createEvent(name: "alarm", value: cmd.value ? "both" : "off")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	def endpoint = cmd.sourceEndPoint
+	if (endpoint == 1) {
+		zwaveEvent(encapsulatedCommand)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	def securedEncapsulatedCommand = cmd.securedEncapsulatedCommand()
+	if (securedEncapsulatedCommand) {
+		zwaveEvent(securedEncapsulatedCommand)
+	} else {
+		log.warn "Unable to extract encapsulated cmd from $cmd"
+		createEvent(descriptionText: cmd.toString())
+	}
+}
+
+private secure(physicalgraph.zwave.Command cmd) {
+	zwave.securityV1.securityMessageEncapsulation().encapsulate(cmd).format()
+}
+
+private onOffCmd(value) {
+    secure(zwave.basicV1.basicSet(value: value))
+}
+
+def on() {
+	resetActiveSound()
+	state.lastTriggeredSound = 1
+	onOffCmd(0xFF)
+}
+
+def off() {
+	state.lastTriggeredSound = 1
+	onOffCmd(0x00)
+}
+
+def strobe() {
+	on()
+}
+
+def siren() {
+	on()
+}
+
+def both() {
+	on()
+}
+
+def ping() {
+	secure(zwave.basicV1.basicGet())
+}
+
+private addChildren(numberOfSounds) {
+	for (def endpoint : 1..numberOfSounds) {
+		try {
+			String childDni = "${device.deviceNetworkId}:$endpoint"
+
+			addChildDevice("Aeotec Child Siren", childDni, device.getHub().getId(), [
+					completedSetup: true,
+					label         : "$device.displayName Sound $endpoint",
+					isComponent   : true,
+					componentName : "sound$endpoint",
+					componentLabel: "Sound $endpoint"
+			])
+		} catch (Exception e) {
+			log.debug "Excep: ${e} "
+		}
+	}
+}
+
+private encap(cmd, endpoint) {
+	if (endpoint) {
+		zwave.multiChannelV3.multiChannelCmdEncap(destinationEndPoint: channelNumber(endpoint)).encapsulate(cmd)
+	} else {
+		cmd
+	}
+}
+
+def channelNumber(String dni) {
+	dni[-1] as Integer
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	if (cmd.notificationStatus == 0xFF) {
+		switch (cmd.event) {
+			case 0x09: //TAMPER
+				createEvent([name: "tamper", value: "detected"])
+				break
+			case 0x01: //ON
+				if (state.lastTriggeredSound == 1) {
+					createEvent([name: "alarm", value: "both"])
+				} else {
+					setActiveSound(state.lastTriggeredSound)
+				}
+				break
+			case 0x00: //OFF
+				resetActiveSound()
+				if (state.lastTriggeredSound == 1) {
+					createEvent([name: "tamper", value: "clear"])
+				}
+				break
+		}
+	}
+}
+
+def setOnChild(deviceDni) {
+	resetActiveSound()
+    sendHubCommand encap(zwave.basicV1.basicSet(value: 0xFF), deviceDni)
+	state.lastTriggeredSound = deviceDni[-1] as Integer
```

Comment:
you've defined a function that does exactly this, but are only using it in one place